library/std/build.rs

```diff
@@ -183,3 +186,48 @@ fn main() {
         println!("cargo:rustc-cfg=reliable_f128_math");
     }
 }
+
+fn build_android() {
+    // Used to detect the value of the `__ANDROID_API__`
+    // builtin #define
+    const MARKER: &str = "BACKTRACE_RS_ANDROID_APIVERSION";
+    const ANDROID_API_C: &str = "
+BACKTRACE_RS_ANDROID_APIVERSION __ANDROID_API__
+";
+
+    // Create `android-api.c` on demand.
+    let out_dir = env::var_os("OUT_DIR").unwrap();
+    let android_api_c = Path::new(&out_dir).join("android-api.c");
```

Comment:
Suggestion: could we leave a remark for *why* this is needed, and if possible link to some docs or references for this? Even if it's just what backtrace-rs I think it'd still be useful.